nixos/modules/services/monitoring/prometheus/xmpp-alerts.nix

```
@@ -4,21 +4,23 @@ with lib;
 
 let
   cfg = config.services.prometheus.xmpp-alerts;
-
-  configFile = pkgs.writeText "prometheus-xmpp-alerts.yml" (builtins.toJSON cfg.configuration);
-
+  settingsFormat = pkgs.formats.yaml {};
+  configFile = settingsFormat.generate "prometheus-xmpp-alerts.yml" cfg.configuration;
 in
-
 {
   options.services.prometheus.xmpp-alerts = {
-
     enable = mkEnableOption "XMPP Web hook service for Alertmanager";
 
     configuration = mkOption {
```

```suggestion
    settings = mkOption {
```

and then add the following to add a migration path between old and new option.

```
  imports = [
    (mkRenamedOptionModule 
      [ "services" "prometheus" "xmpp-alerts" "configuration" ]
      [ "services" "prometheus" "xmpp-alerts" "settings" ])
  ];
```